src/algebra/monoid_algebra.lean

```diff
@@ -638,20 +628,11 @@ section algebra
 variables {R : Type*}
 local attribute [reducible] add_monoid_algebra
 
-/--
-As a preliminary to defining the `k`-algebra structure on `add_monoid_algebra k G`,
-we define the underlying ring homomorphism.
-
-In fact, we do this in more generality, providing the ring homomorphism
-`R →+* add_monoid_algebra k G` given any ring homomorphism `R →+* k`.
--/
-def algebra_map' [semiring R] [semiring k] (f : R →+* k) [add_monoid G] :
-  R →+* add_monoid_algebra k G :=
-{ to_fun := λ x, single 0 (f x),
-  map_one' := by { simp, refl },
-  map_mul' := λ x y, by rw [single_mul_single, zero_add, f.map_mul],
-  map_zero' := by rw [f.map_zero, single_zero],
-  map_add' := λ x y, by rw [f.map_add, single_add], }
+/-- `finsupp.single 0` as a `ring_hom` -/
+@[simps] def single_zero_ring_hom [semiring k] [add_monoid G] : k →+* add_monoid_algebra k G :=
```

Comment:
```suggestion
@[simps] def single_zero_hom [semiring k] [add_monoid G] : k →+* add_monoid_algebra k G :=
```

Reply:
See below - with `single_zero_alg_hom` present, this seems undesirable.